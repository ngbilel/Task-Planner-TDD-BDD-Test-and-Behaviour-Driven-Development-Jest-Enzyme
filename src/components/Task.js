import React, {Component}  from 'react'
import  {Form, FormGroup, FormControl, ControlLabel, Button} from 'react-bootstrap'
class   Task extends Component {
    constructor(props){
        super(props);
        this.state={
            person: '',
            task:''
        }
    }

    render(){
        return (
            <div>
                <Form>
                    <FormGroup>
                        <FormControl className='input-person' onChange={(e)=>this.setState({person:e.target.value})} />
                    </FormGroup>
                </Form>
            </div>
        )     

    }      
}

export default Task