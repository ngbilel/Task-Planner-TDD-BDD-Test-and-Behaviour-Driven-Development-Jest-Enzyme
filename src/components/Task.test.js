import React from 'react';
import {shallow} from 'enzyme';
import Task from './Task';

describe('Task',()=>{
    const task = shallow (<Task/>);
    const person ='Teddy';
    
    it('Render porperly',()=>{
        expect(task).toMatchSnapshot();
    })

    it('initializes a person and task in `state`',()=>{
        expect(task.state()).toEqual({person: '', task:''})
    })

    describe('When typing in person input',()=>{
        beforeEach(()=>{
            task.find('.input-person').simulate('change',{target:{value:person}})
        })

        it('updates the person in `state`',()=>{
            expect(task.state().person).toEqual(person)
        })
    })


})

