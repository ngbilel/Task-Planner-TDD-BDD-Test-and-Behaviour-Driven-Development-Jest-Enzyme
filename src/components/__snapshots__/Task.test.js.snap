// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Render porperly 1`] = `ShallowWrapper {}`;
